RubeesEat/Model/InMemoryBillRepository.cs

```diff
@@ -29,4 +29,11 @@ public void Add(Bill bill)
         ArgumentNullException.ThrowIfNull(bill);
         _bills.Add(bill.Id, bill);
     }
+
+    public decimal GetBalance(Guid personId)
+    {
+        return _bills.Values.Sum(bill => bill.EntryLines
+                                             .Where(entryLine => entryLine.Person.Id == personId)
+                                             .Sum(entryLine => entryLine.Amount));
```

Comment:
Alternative LINQ expression using `.SelectMany(...)` (`flatMap` in JS) as an example:

```c#
return _bills.Values
    .SelectMany(e => e.EntryLines)
    .Where(e => e.Person == person)
    .Sum(e => e.Amount);
```

`SelectMany` selects an `IEnumerable<T>` and "flattens" it out so that the elements for a single enumerable. So we change the enumeration from `Bill` to `EntryLine` because we don't care about the details of the `Bill`. Then we filter and sum just on an enumerable of `EntryLine`.

Example from JS, which might convey the idea a bit easier: `[[1, 3], 4, [5, 6]].flatMap(e => e) ` -> `[ 1, 3, 4, 5, 6 ]`